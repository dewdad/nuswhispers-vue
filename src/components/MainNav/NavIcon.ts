import Vue from 'vue';
import Component from 'vue-class-component';

@Component({
  props: {
    external: Boolean,
    icon: String,
    to: String,
  },
})
export default class NavIcon extends Vue {
}
