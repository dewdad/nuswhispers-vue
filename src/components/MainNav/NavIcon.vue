<template>
  <a v-if="external" :href="to" target="_blank" rel="noopener noreferrer">
    <icon :id="icon" :class="$style.icon" />
  </a>
  <router-link v-else :to="to">
    <icon :id="icon" :class="$style.icon" />
  </router-link>
</template>

<style module>
.icon {
  fill: rgba(255, 255, 255, .5);
  height: 28px;
  transition: fill .2s;
  width: 28px;
  vertical-align: -40%;
}

.icon:hover {
  fill: #fff;
}
</style>

<script lang="ts" src="./NavIcon.ts" />
