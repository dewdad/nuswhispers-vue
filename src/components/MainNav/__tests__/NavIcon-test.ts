import Vue from 'vue';
import { createRenderer } from 'vue-server-renderer';
import NavIcon from '../NavIcon.vue';

describe('NavIcon', () => {
  const renderer = createRenderer();

  it('renders correctly', () => {
    const vm = new Vue({
      propsData: {
        icon: 'zoom-outline',
        to: '/search',
      },
      render: (h) => h(NavIcon),
    });
    renderer.renderToString(vm, (err: Error | undefined, html: string) => {
      expect(html).toMatchSnapshot();
    });
  });
});
