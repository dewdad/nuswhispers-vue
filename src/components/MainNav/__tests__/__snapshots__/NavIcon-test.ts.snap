// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIcon renders correctly 1`] = `
<router-link data-server-rendered="true">
    <icon class="icon"></icon>
</router-link>
`;
